File: internal/content/testdata/bench/javascript/nc-reverse-k-group.js
function reverse_k_group(head, k) {
    function getKth(cur, k) {
        while (cur !== null && k > 0) {
            cur = cur.next;
            k--;
        }
        return cur;
    }
    const dummy = new ListNode(0, head);
    let groupPrev = dummy;
    while (true) {
        const kth = getKth(groupPrev, k);
        if (kth === null) break;
        const groupNext = kth.next;
        let prev = groupNext, cur = groupPrev.next;
        while (cur !== groupNext) {
            const nxt = cur.next;
            cur.next = prev;
            prev = cur;
            cur = nxt;
        }
        const tmp = groupPrev.next;
        groupPrev.next = kth;
        groupPrev = tmp;
    }
    return dummy.next;
}

File: internal/content/testdata/bench/rust/nc-reverse-k-group.rs
fn reverse_k_group(head: Option<Box<ListNode>>, k: i64) -> Option<Box<ListNode>> {
    // Extract values into a Vec.
    let mut vals: Vec<i32> = Vec::new();
    let mut cur = &head;
    while let Some(n) = cur {
        vals.push(n.val);
        cur = &n.next;
    }
    // Reverse each full group of k.
    let kk = k as usize;
    if kk > 0 {
        let len = vals.len();
        let mut i = 0;
        while i + kk <= len {
            vals[i..i + kk].reverse();
            i += kk;
        }
    }
    // Rebuild the list.
    let mut node: Option<Box<ListNode>> = None;
    for &v in vals.iter().rev() {
        let mut b = Box::new(ListNode::new(v));
        b.next = node;
        node = Some(b);
    }
    node
}
